Mathlib/Topology/UniformSpace/UniformConvergence.lean

```diff
@@ -351,6 +351,66 @@ theorem UniformContinuous₂.tendstoUniformly [UniformSpace α] [UniformSpace γ
     (h : UniformContinuous₂ f) : TendstoUniformly f (f x) (𝓝 x) :=
   UniformContinuousOn.tendstoUniformly univ_mem <| by rwa [univ_prod_univ, uniformContinuousOn_univ]
 
+
+namespace Filter.HasBasis
+
+variable {X ιX ια ιβ : Type*}
+
+/-- An anologue of `Filter.HasBasis.tendsto_right_iff` for `TendstoUniformlyOnFilter`. -/
+lemma uniformity_tendstoUniformlyOnFilter_iff {F : X → α → β} {f : α → β}
+    {l : Filter X} {l' : Filter α} {pβ : ιβ → Prop} {sβ : ιβ → Set (β × β)}
+    (hβ : (uniformity β).HasBasis pβ sβ) :
+    TendstoUniformlyOnFilter F f l l' ↔
+      (∀ i, pβ i → ∀ᶠ n in l ×ˢ l', (f n.2, F n.1 n.2) ∈ sβ i) := by
+  rw [TendstoUniformlyOnFilter, hβ.forall_iff]
+  exact fun _ _ _ h ↦ h.mp <| .of_forall <| by aesop
+
+/-- An anologue of `Filter.HasBasis.tendsto_right_iff` for `TendstoUniformlyOn`. -/
+lemma tendstoUniformlyOnFilter_iff {F : X → α → β} {f : α → β}
+    {l : Filter X} {l' : Filter α} {pX : ιX → Prop} {sX : ιX → Set X}
+    {pα : ια → Prop} {sα : ια → Set α} {pβ : ιβ → Prop} {sβ : ιβ → Set (β × β)}
+    (hl : l.HasBasis pX sX) (hl' : l'.HasBasis pα sα)
+    (hβ : (uniformity β).HasBasis pβ sβ) :
+    TendstoUniformlyOnFilter F f l l' ↔
+      (∀ i, pβ i → ∃ j k, (pX j ∧ pα k) ∧ ∀ x a, x ∈ sX j → a ∈ sα k → (f a, F x a) ∈ sβ i) := by
+  simp [hβ.uniformity_tendstoUniformlyOnFilter_iff, (hl.prod hl').eventually_iff]
```

Comment:
I'm extremely surprised that we don't have these, but I can't find them and the proof of [EMetric.tendstoLocallyUniformly_iff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/EMetricSpace/Basic.html#EMetric.tendstoLocallyUniformly_iff) seems to point in that direction.

Reply:
I could not find them either.